app/policies/annotation_policy.rb

```diff
@@ -0,0 +1,5 @@
+class AnnotationPolicy < ApplicationPolicy
+  def add_existing_annotation?
+    user.admin? || (user.ta? && GraderPermission.find_by(user_id: user.id).create_delete_annotations)
```

Comment:
You can probably simplify this syntax by making a GraderPermissionPolicy class and defining a policy for each configuration option.